giftless/transfer/basic_streaming.py

```diff
@@ -18,93 +16,12 @@
 from giftless.auth.identity import Permission
 from giftless.exc import InvalidPayload, NotFound
 from giftless.schema import ObjectSchema
-from giftless.transfer import PreAuthorizingTransferAdapter, ViewProvider, exc
+from giftless.storage import StreamingStorage, VerifiableStorage
+from giftless.transfer import PreAuthorizingTransferAdapter, ViewProvider
 from giftless.util import get_callable
 from giftless.view import BaseView
 
 
-class VerifiableStorage(ABC):
-    """A storage backend that supports object verification API
-
-    All streaming backends should be 'verifiable'.
-    """
-    def verify_object(self, prefix: str, oid: str, size: int) -> bool:
-        """Check that object exists and has the right size
-
-        This method should not throw an error if the object does not exist, but return False
-        """
-        pass
-
-
-class StreamingStorage(VerifiableStorage, ABC):
```

Comment:
I think that the basic interface for a streaming storage should stay here in this new layout.